chore: add python CI Github Action

tzellman opened the pull request.

dpressel commented:
Looks good, but I do think we are going to want a few different OS builds, might want to make the file name something alluding to it being `ubuntu`

tzellman commented:
Sure! I'm not 100% sure what Github supports, but I can take a look.

I just added it so that there was some sort of CI build for running tests (I added flake8 as well, but can remove that if you want).

The other node branch/PR has an action to build the node.js bindings, and run those tests as well.

dpressel commented:
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-software

was thinking macOS at a minimum
